src/main/java/io/kestra/plugin/notifications/services/ExecutionService.java

```diff
@@ -101,12 +98,24 @@ public static Map<String, Object> executionMap(RunContext runContext, ExecutionI
             templateRenderMap.put("customFields", renderedCustomFields);
         }
 
-        TaskRun lastTaskRun = execution.getTaskRunList().stream()
-            .filter(t -> (execution.hasFailed() ? State.Type.FAILED : State.Type.SUCCESS).equals(t.getState().getCurrent()))
-            .toList()
-            .getLast();
+        var executionVars = (Map<String, String>) runContext.getVariables().get("execution");
+        var isCurrentExecution = execution.getId().equals(executionVars.get("id"));
+        var taskRuns = execution.getTaskRunList();
+
+        final TaskRun lastTaskRun;
+        var lastTaskRunState = execution.getTaskRunList().getLast().getState().getCurrent();
+
+        if (isCurrentExecution && State.Type.RUNNING.equals(lastTaskRunState)) {
```

Comment:
looks good, I think it could be nice to make a function isCurrentExecution() and reuse it in findExecution